File: mobile/lib/dtos/card_stats_update_request.dart
class CardStatsUpdateRequest {
  int cardId;
  num easeFactor;
  int interval;
  int repetitions;
  DateTime dueDate;

  CardStatsUpdateRequest({
    required this.cardId,
    required this.easeFactor,
    required this.interval,
    required this.repetitions,
    required this.dueDate,
  });

  factory CardStatsUpdateRequest.fromJson(Map<String, dynamic> json) =>
      CardStatsUpdateRequest(
        cardId: json["cardId"],
        easeFactor: json["easeFactor"],
        interval: json["interval"],
        repetitions: json["repetitions"],
        dueDate: DateTime.parse(json["dueDate"]),
      );

  Map<String, dynamic> toJson() => {
        "cardId": cardId,
        "easeFactor": easeFactor,
        "interval": interval,
        "repetitions": repetitions,
        "dueDate": dueDate.toIso8601String(),
      };
}

File: backend/MemoWise.Model/DTO/CardStatsUpdateRequest.cs
namespace MemoWise.Model.DTO;

public class CardStatsUpdateRequest
{
    public int CardId { get; set; }
    public float EaseFactor { get; set; } 
    public int Interval { get; set; }
    public int Repetitions { get; set; }
    public DateTime DueDate { get; set; }

    public CardStatsUpdateRequest() {}

    public CardStatsUpdateRequest(int id, float easeFactor, int interval, int repetitions, DateTime dueDate)
    {
        CardId = id;
        EaseFactor = easeFactor;
        Interval = interval;
        Repetitions = repetitions;
        DueDate = dueDate;
    }
}
